docs/languages/csharp/properties.md

```diff
@@ -0,0 +1,255 @@
+                throw new ArgumentException("First name must not be blank");
+            firstName = value;
+        }
+    }
+    private string firstName;
+    // remaining implementation removed from listing
+}
+```
+
+The example above enforces the rule that the first name must not be blank,
+or whitespace. If a developer writes
+```
+foo.FirstName = "";
+```
+That assignment throws an `ArgumentException`. Because a property set accessor
+must have a void return type, you report errors in the set accessor by throwing an exception.
+
+That is a simple case of validation. You can extend this same syntax to anything needed
+in your scenario: You can check the relationships between different properties, or validate
+against any external conditions. Any valid C# statements are valid in a property accessor.
+
+## Readonly
+
+Up to this point, all the property definitions you have seen are read-write properties
+with public accessors. That's not the only valid accessibility for properties.
+You can create readonly properties, or give different accessibility to the set and get
+accessors. Suppose that your `Person` class should only enable changing the value of the
+FirstName property from other methods in that class. You could give the set accessor
+`private` accessibility instead of `public`:
+
+```
+public class Person
+{
+	public string FirstName
+	{
+		get;
+		private set;
+	}
+	// remaining implementation removed from listing
+}
+```
+
+Now, the `FirstName` property can be accessed from any code, but it can only be assigned
+from other code in the `Person` class.
+You can add any restrictive access modifier to either the set or get accessors. Any access modifier
+you place on the individual accessor must be more limited than the access modifier on the property
+definition. The above is legal because the `FirstName` property is `public`, but the set accessor is
+`private`. You could not declare a `private` property with a `public` accessor. Property declarations
+can also be declared `protected`, `internal`, `protected internal` or even `private`.   
+
+It is also legal to place the more restrictive modifier on the `get` accessor. For example, you could
+have a `public` property, but restrict the `get` accessor to `private`. That scenario is rarely done
+in practice.
+ 
+## Computed Properties
+
+A property does not need to simply return the value of a member field. You can create properties
+that return a computed value. Let's expand the `Person` object to return the fulll name, computed
+by concatenating the first and last names:
+
+```
+public class Person
+{
+    public string FirstName
+    {
+        get;
+        set;
+    }
+
+    public string LastName
+    {
+        get;
+        set;
+    }
+
+    public string FullName
+    {
+        get
+        {
+            return $"{FirstName} {LastName}";
+        }
+    }
+}
+```
+
+The example above uses the new *String Interpolation* syntax to create a 
```

Comment:
I'd avoid the use of "new" here.  We'd like these samples to reflect the latest and greatest way to do things except for in cases where we're providing a sample for an older version of .NET where Roslyn features can't be used.